### ISSUE TITLE
Kaniko with gVisor: Failed to load /busybox/sh

### ISSUE THREAD

**Issue by MichelMan** [GoogleContainerTools/kaniko#985]
**Description**

I would like to build a Dockerfile with Kaniko in gVisor. So after read the documentation in [gVisor](https://gvisor.dev/docs/user_guide/docker/), i installed runsc and configured it and read the documentation to use Kaniko as describe in [the section of gVisor](https://github.com/GoogleContainerTools/kaniko#running-kaniko-in-gvisor).

In my case, i want to use **Kaniko with gVisor in GitLab CI/CD**. So for this reason, i used the **kaniko debug image (gcr.io/kaniko-project/executor:debug)** recommanded in [Building a Docker image with kaniko](https://docs.gitlab.com/ee/ci/docker/using_kaniko.html). For some tests, i tried to use it with docker first.
``` 
$ sudo docker run --runtime=runsc --entrypoint="" -it -v $(pwd):/workspace –v \  /root/.docker/config.json:/kaniko/.docker/config.json gcr.io/kaniko-project/executor:debug sh
```
In result of this command:
```
docker: Error response from daemon: OCI runtime start failed: starting container: starting root container: starting sandbox: creating init process: Failed to load /busybox/sh: format error: unknown
```
I'm new in this domain, so i wonder if its possible to execute gVisor with kaniko debug image for using shell for job GitLab CI/CD.

I hope to have more explication on this subject. Thanks in advance.

**Comment by tejal29** [comment]
Can you try 
```
$ sudo docker run --runtime=runsc --entrypoint="" -it -v $(pwd):/workspace –v \  /root/.docker/config.json:/kaniko/.docker/config.json gcr.io/kaniko-project/executor:debug sh /busybox/sh
```

**Comment by MichelMan** [comment]
Same result with your command:
```
$ docker run --runtime=runsc --entrypoint=/busybox/sh -it -v $(pwd):/workspace -v /root/.docker/config.json:/kaniko/.docker/config.json gcr.io/kaniko-project/executor:debug
```

```
docker: Error response from daemon: OCI runtime start failed: starting container: starting root container: starting sandbox: creating init process: Failed to load /busybox/sh: exec format error: unknown.
ERRO[0000] error waiting for container: context canceled
```

Default behavior with Docker (runc):
```
$ docker run --runtime=runsc --entrypoint=/busybox/sh -it -v $(pwd):/workspace -v /root/.docker/config.json:/kaniko/.docker/config.json gcr.io/kaniko-project/executor:debug
/ # (shell)
```

**Comment by tejal29** [comment]
i don't have `runsc` installed.
Kaniko documentation mentions https://github.com/GoogleContainerTools/kaniko#running-kaniko-in-gvisor using the `--force` flag for the executor image. If this works, i will install gvisor and try on my end.

**Comment by MichelMan** [comment]
Yes I already tried it with Kaniko documentation. It _**seems to work**_ with Docker and Kaniko image `gcr.io/kaniko-project/executor:latest` with the `--force` flag (Cf. issues https://github.com/GoogleContainerTools/kaniko/issues/789).

But in this case, i tried to build a Dockerfile with Kaniko debug image `gcr.io/kaniko-project/executor:debug` with `runsc` in order to use it in Gitlab CI (where we have to use the shell so Kaniko debug image Cf. https://docs.gitlab.com/ee/ci/docker/using_kaniko.html#building-a-docker-image-with-kaniko).

**Comment by invliD** [comment]
I'm seeing the same issue running kaniko through gitlab ci on a containerd-based kubernetes cluster.

The root cause seems to be gvisor not supporting 32bit binaries, and busybox as included in distroless (which the kaniko image is based on) is a 32bit binary: GoogleContainerTools/distroless#352

**Comment by MichelMan** [comment]
Thanks @invliD, i thought i was the only one having this problem.
It's a little more clear now. So we have to wait for a fix on the side of the Kaniko debug image...

**Comment by alex1989hu** [comment]
We are also affected - upvote +1

**Comment by invliD** [comment]
This was fixed with #1254, released in v0.23.0. I can run the v1.2.0 debug image successfully using gVisor.

**Comment by tejal29** [comment]
Thanks @invliD for confirming the fix.
